class-astra-theme-customizer-reset.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Plugin Name: Astra Customizer Reset
+ * Plugin URI: http://www.ultimatebeaver.com/
+ * Description: Reset astra theme customizer options.
+ * Version: 1.0.0
+ * Author: Brainstorm Force
+ * Author URI: http://www.brainstormforce.com
+ * Text Domain: astra-customizer-reset
+ *
+ * @package Astra Customizer Reset
+ */
+
+define( 'ASTRA_CUSTOMIZER_RESET_URI', plugins_url( '/', __FILE__ ) );
+
+/**
+ * Astra Customizer Reset
+ *
+ * @since 1.0.0
+ */
+if ( ! class_exists( 'Astra_Theme_Customizer_Reset' ) ) {
+
+	/**
+	 * Astra Customizer Reset
+	 */
+	class Astra_Theme_Customizer_Reset {
+
+		/**
+		 * Member Variable
+		 *
+		 * @since 1.0.0
+		 * @var instance
+		 */
+		private static $instance;
+
+		/**
+		 * Initiator
+		 *
+		 * @since 1.0.0
+		 */
+		public static function get_instance() {
+			if ( ! isset( self::$instance ) ) {
+				self::$instance = new self;
+			}
+			return self::$instance;
+		}
+
+		/**
+		 * Constructor
+		 *
+		 * @since 1.0.0
+		 */
+		public function __construct() {
+
+			add_action( 'wp_ajax_astra_customizer_reset',  			array( $this, 'ajax_customizer_reset' ) );
+			add_action( 'customize_controls_enqueue_scripts',   	array( $this, 'controls_scripts' ) );
+
+		}
+
+		/**
+		 * AJAX Customizer Reset
+		 *
+		 * @since 1.0.0
+		 * @return void
+		 */
+		public function ajax_customizer_reset() {
+
+			// Validate nonce.
+			check_ajax_referer( 'astra-customizer-reset', 'nonce' );
+
+			// Reset option 'astra-settings'.
```

Comment:
Added `$this->wp_customize->is_preview()` condition before removing customizer options.